sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JdbcUtils.scala

```diff
@@ -246,13 +247,23 @@ object JdbcUtils extends Logging {
   }
 
   /**
+   * The utility to add quote to the column name based on its dialect
+   * @param dialect the JDBC dialect
+   * @param columnName the input column name
+   * @return the quoted column name
+   */
+  private def quoteColumnName(dialect: JdbcDialect, columnName: String): String = {
+    dialect.quoteIdentifier(columnName)
+  }
+
+  /**
    * Compute the schema string for this RDD.
    */
-  def schemaString(df: DataFrame, url: String): String = {
+  def schemaString(dialect: JdbcDialect, df: DataFrame, url: String): String = {
     val sb = new StringBuilder()
     val dialect = JdbcDialects.get(url)
```

Comment:
You pass in a parameter named dialect to the schemaString method, but there's also the dialect that comes from JdbcDialects.get(url) --- that's the duplicate I was trying to point out